README.md

```diff
@@ -25,7 +25,53 @@ The following tool(s) must be installed on your system prior to developing the w
 We also assume you've got GNU coreutils installed, which is usually the case by default on Linux
 and should be installable via Homebrew on macOS.
 
-## Website Development
+## Website Development Guides
+
+### Where's the documentation content?
+
+All docs go under `content/`; most changes are usually made in `content/docs` which is conceptually
+the "latest" version of the docs and includes other sections which don't make sense to version. For
+example, "contributing" and "supported releases" are only available under `content/docs/` because
+people should only want to consume the latest versions of those pages, and their content isn't
+tied to a cert-manager version.
+
+Other folders under `content/` refer to specific versions of cert-manager; e.g. v1.8-docs is for
+cert-manager 1.8.
+
+### Task: Linking to other documentation pages
+
+When working on documentation, all links to other documentation pages should be _relative_ to the
+document you're working on and should point at a named markdown file.
+
+For example, take [this snapshot](https://raw.githubusercontent.com/cert-manager/website/d398905baef9841590fab6c2b854b74f0eecb006/content/docs/concepts/certificate.md)
+of the "Certificate Concepts" page.
+
+There's an external link to an RFC which is fully specified as expected:
+
+> `see [RFC 5246 section 7.4.2](https://datatracker.ietf.org/doc/html/rfc5246#section-7.4.2)`
+
+But towards the end of the page we link to the "Certificate Usage" page:
```

Comment:
Praise: I've previously spent ages searching through the docs for an example of how to link a file in the same repo as I always forget how to do that. Adding this example is absolutely awesome